types/knockout.mapping/index.d.ts

```diff
@@ -13,8 +14,14 @@ declare var self: KnockoutMapping;
 export = self;
 
 declare global {
-    type KnockoutObservableType<T> = {	
-        [P in keyof T]: KnockoutObservable<KnockoutObservableType<T[P]>>|T[P];	
+	type KnockoutObservableType<T> = {
```

Comment:
I'm not really familiar with `knockout.mapping`, but by looking at type definition I can suggest
- extract conditional type to another helper type
- test for primitive types first without going into mapped type

something like:

```ts
type KnockoutObservableType<T> =
  T extends Date | string | boolean | number ? KnockoutObservable<T> :
  T extends (infer E)[] ? KnockoutObservableArray<KnockoutObservableType<E>> :
  T extends object ? { [P in keyof T]: KnockoutObservableType<T[P]> } :
  unknown;
```

Reply:
That is really pretty, but I get `error TS2456: Build:Type alias 'KnockoutObservableType' circularly references itself.` from it. Taking the good part of the comment above, I end up with
```typescript
type KnockoutObservableType<T> = {
    [P in keyof T]:
    T[P] extends Date | string | boolean | number ? KnockoutObservable<T[P]> :
    T[P] extends (infer E)[] ? KnockoutObservableArray<KnockoutObservableType<E>> :
    T[P] extends object ? T[P] :
    unknown;
};
```

Better?